# Indian Premier League : Sports analysis 

# Phase 1 : data cleaning

In [1]:
import pandas as pd
import numpy as np
# Read the CSV file from the URL into a pandas dataframe with 'ISO-8859-1' encoding
url = 'https://query.data.world/s/2a6soodhpibej6iu67a2upte7euehy?dws=00000'
player_match_df = pd.read_csv(url, encoding='ISO-8859-1')

# Display the dataframe
player_match_df.head()

Player_match_SK  PlayerMatch_key  Match_Id  Player_Id Player_Name  \
0            12694      33598700006    335987          6    R Dravid   
1            12695      33598700007    335987          7    W Jaffer   
2            12696      33598700008    335987          8     V Kohli   
3            12697      33598700009    335987          9   JH Kallis   
4            12698      33598700010    335987         10    CL White   

          DOB    Batting_hand          Bowling_skill  Country_Name Role_Desc  \
0   1/11/1973  Right-hand bat     Right-arm offbreak         India   Captain   
1   2/16/1978  Right-hand bat     Right-arm offbreak         India    Player   
2   11/5/1988  Right-hand bat       Right-arm medium         India    Player   
3  10/16/1975  Right-hand bat  Right-arm fast-medium  South Africa    Player   
4   8/18/1983  Right-hand bat        Legbreak googly     Australia    Player   

   ... Season_year is_manofThematch  Age_As_on_match  IsPlayers_Team_won  \
0  ...        2008            False               35               False   
1  ...        2008            False               30               False   
2  ...        2008            False               20               False   
3  ...        2008            False               33               False   
4  ...        2008            False               25               False   

   Batting_Status  Bowling_Status  Player_Captain  Opposit_captain  \
0             NaN             NaN        R Dravid       SC Ganguly   
1             NaN             NaN        R Dravid       SC Ganguly   
2             NaN             NaN        R Dravid       SC Ganguly   
3             NaN             NaN        R Dravid       SC Ganguly   
4             NaN             NaN        R Dravid       SC Ganguly   

  Player_keeper Opposit_keeper  
0    MV Boucher        WP Saha  
1    MV Boucher        WP Saha  
2    MV Boucher        WP Saha  
3    MV Boucher        WP Saha  
4    MV Boucher        WP Saha  

[5 rows x 22 columns]

In [2]:
url2 = 'https://query.data.world/s/zsvjf3datludh4ih4efoxtieqjmq4f?dws=00000'
match_df = pd.read_csv(url2, encoding='ISO-8859-1')

# Display the dataframe
match_df

Match_SK  match_id                        Team1  \
0           0    335987  Royal Challengers Bangalore   
1           1    335988              Kings XI Punjab   
2           2    335989             Delhi Daredevils   
3           3    335990               Mumbai Indians   
4           4    335991        Kolkata Knight Riders   
..        ...       ...                          ...   
632       632   1082646             Delhi Daredevils   
633       633   1082647               Mumbai Indians   
634       634   1082648          Sunrisers Hyderabad   
635       635   1082649               Mumbai Indians   
636       636   1082650               Mumbai Indians   

                           Team2 match_date  Season_Year  \
0          Kolkata Knight Riders  4/18/2008         2008   
1            Chennai Super Kings  4/19/2008         2008   
2               Rajasthan Royals  4/19/2008         2008   
3    Royal Challengers Bangalore  4/20/2008         2008   
4                Deccan Chargers  4/20/2008         2008   
..                           ...        ...          ...   
632  Royal Challengers Bangalore  5/14/2017         2017   
633      Rising Pune Supergiants  5/16/2017         2017   
634        Kolkata Knight Riders  5/17/2017         2017   
635        Kolkata Knight Riders  5/19/2017         2017   
636      Rising Pune Supergiants  5/21/2017         2017   

                                     Venue_Name           City_Name  \
0                         M Chinnaswamy Stadium           Bangalore   
1    Punjab Cricket Association Stadium, Mohali          Chandigarh   
2                              Feroz Shah Kotla               Delhi   
3                              Wankhede Stadium              Mumbai   
4                                  Eden Gardens             Kolkata   
..                                          ...                 ...   
632                            Feroz Shah Kotla               Delhi   
633                            Wankhede Stadium              Mumbai   
634                       M Chinnaswamy Stadium           Bengaluru   
635                       M Chinnaswamy Stadium           Bengaluru   
636    Rajiv Gandhi International Stadium Uppal  Hyderabad (Deccan)   

    Country_Name                  Toss_Winner                 match_winner  \
0          India  Royal Challengers Bangalore        Kolkata Knight Riders   
1          India          Chennai Super Kings          Chennai Super Kings   
2          India             Rajasthan Royals             Delhi Daredevils   
3          India               Mumbai Indians  Royal Challengers Bangalore   
4          India              Deccan Chargers        Kolkata Knight Riders   
..           ...                          ...                          ...   
632        India  Royal Challengers Bangalore  Royal Challengers Bangalore   
633        India               Mumbai Indians      Rising Pune Supergiants   
634        India        Kolkata Knight Riders        Kolkata Knight Riders   
635        India               Mumbai Indians               Mumbai Indians   
636        India               Mumbai Indians               Mumbai Indians   

    Toss_Name Win_Type Outcome_Type         ManOfMach  Win_Margin  Country_id  
0       field     runs       Result       BB McCullum       140.0           1  
1         bat     runs       Result        MEK Hussey        33.0           1  
2         bat  wickets       Result       MF Maharoof         9.0           1  
3         bat  wickets       Result        MV Boucher         5.0           1  
4         bat  wickets       Result         DJ Hussey         5.0           1  
..        ...      ...          ...               ...         ...         ...  
632       Bat     runs       Result          HV Patel        10.0           1  
633     Field     runs       Result  Wasington Sundar        20.0           1  
634     Field  wickets       Result   NM Coulter-Nile         7.0           1  
635     Field  wicket

In [3]:
player_match_df = player_match_df.rename(columns={'Match_Id': 'match_id'})
player_match_df.head()
#renaming the column on which we are merging the datasets

Player_match_SK  PlayerMatch_key  match_id  Player_Id Player_Name  \
0            12694      33598700006    335987          6    R Dravid   
1            12695      33598700007    335987          7    W Jaffer   
2            12696      33598700008    335987          8     V Kohli   
3            12697      33598700009    335987          9   JH Kallis   
4            12698      33598700010    335987         10    CL White   

          DOB    Batting_hand          Bowling_skill  Country_Name Role_Desc  \
0   1/11/1973  Right-hand bat     Right-arm offbreak         India   Captain   
1   2/16/1978  Right-hand bat     Right-arm offbreak         India    Player   
2   11/5/1988  Right-hand bat       Right-arm medium         India    Player   
3  10/16/1975  Right-hand bat  Right-arm fast-medium  South Africa    Player   
4   8/18/1983  Right-hand bat        Legbreak googly     Australia    Player   

   ... Season_year is_manofThematch  Age_As_on_match  IsPlayers_Team_won  \
0  ...        2008            False               35               False   
1  ...        2008            False               30               False   
2  ...        2008            False               20               False   
3  ...        2008            False               33               False   
4  ...        2008            False               25               False   

   Batting_Status  Bowling_Status  Player_Captain  Opposit_captain  \
0             NaN             NaN        R Dravid       SC Ganguly   
1             NaN             NaN        R Dravid       SC Ganguly   
2             NaN             NaN        R Dravid       SC Ganguly   
3             NaN             NaN        R Dravid       SC Ganguly   
4             NaN             NaN        R Dravid       SC Ganguly   

  Player_keeper Opposit_keeper  
0    MV Boucher        WP Saha  
1    MV Boucher        WP Saha  
2    MV Boucher        WP Saha  
3    MV Boucher        WP Saha  
4    MV Boucher        WP Saha  

[5 rows x 22 columns]

In [4]:
# Merge the datasets using match_id column
merged_df = pd.merge(player_match_df, match_df, on='match_id')

# Print the merged dataframe
merged_df.head()

Player_match_SK  PlayerMatch_key  match_id  Player_Id Player_Name  \
0            12694      33598700006    335987          6    R Dravid   
1            12695      33598700007    335987          7    W Jaffer   
2            12696      33598700008    335987          8     V Kohli   
3            12697      33598700009    335987          9   JH Kallis   
4            12698      33598700010    335987         10    CL White   

          DOB    Batting_hand          Bowling_skill Country_Name_x Role_Desc  \
0   1/11/1973  Right-hand bat     Right-arm offbreak          India   Captain   
1   2/16/1978  Right-hand bat     Right-arm offbreak          India    Player   
2   11/5/1988  Right-hand bat       Right-arm medium          India    Player   
3  10/16/1975  Right-hand bat  Right-arm fast-medium   South Africa    Player   
4   8/18/1983  Right-hand bat        Legbreak googly      Australia    Player   

   ...  City_Name Country_Name_y                  Toss_Winner  \
0  ...  Bangalore          India  Royal Challengers Bangalore   
1  ...  Bangalore          India  Royal Challengers Bangalore   
2  ...  Bangalore          India  Royal Challengers Bangalore   
3  ...  Bangalore          India  Royal Challengers Bangalore   
4  ...  Bangalore          India  Royal Challengers Bangalore   

            match_winner  Toss_Name  Win_Type  Outcome_Type    ManOfMach  \
0  Kolkata Knight Riders      field      runs        Result  BB McCullum   
1  Kolkata Knight Riders      field      runs        Result  BB McCullum   
2  Kolkata Knight Riders      field      runs        Result  BB McCullum   
3  Kolkata Knight Riders      field      runs        Result  BB McCullum   
4  Kolkata Knight Riders      field      runs        Result  BB McCullum   

  Win_Margin Country_id  
0      140.0          1  
1      140.0          1  
2      140.0          1  
3      140.0          1  
4      140.0          1  

[5 rows x 38 columns]

In [5]:
# Count the number of null values in each column
null_count = merged_df.isnull().sum()

# Print the results
print(null_count)

Player_match_SK           0
PlayerMatch_key           0
match_id                  0
Player_Id                 0
Player_Name               0
DOB                       0
Batting_hand              0
Bowling_skill          1130
Country_Name_x            0
Role_Desc                 0
Player_team               0
Opposit_Team              0
Season_year               0
is_manofThematch          0
Age_As_on_match           0
IsPlayers_Team_won        0
Batting_Status        13992
Bowling_Status        13992
Player_Captain            0
Opposit_captain           0
Player_keeper             0
Opposit_keeper            0
Match_SK                  0
Team1                     0
Team2                     0
match_date                0
Season_Year               0
Venue_Name                0
City_Name                 0
Country_Name_y            0
Toss_Winner               0
match_winner             66
Toss_Name                 0
Win_Type                 22
Outcome_Type              0
ManOfMach           

In [6]:
# Drop the 'Batting_Status' and 'Bowling_Status' columns from the DataFrame
clean_df = merged_df.drop(['Batting_Status', 'Bowling_Status'], axis=1)

# Print the modified DataFrame
clean_df.head()

Player_match_SK  PlayerMatch_key  match_id  Player_Id Player_Name  \
0            12694      33598700006    335987          6    R Dravid   
1            12695      33598700007    335987          7    W Jaffer   
2            12696      33598700008    335987          8     V Kohli   
3            12697      33598700009    335987          9   JH Kallis   
4            12698      33598700010    335987         10    CL White   

          DOB    Batting_hand          Bowling_skill Country_Name_x Role_Desc  \
0   1/11/1973  Right-hand bat     Right-arm offbreak          India   Captain   
1   2/16/1978  Right-hand bat     Right-arm offbreak          India    Player   
2   11/5/1988  Right-hand bat       Right-arm medium          India    Player   
3  10/16/1975  Right-hand bat  Right-arm fast-medium   South Africa    Player   
4   8/18/1983  Right-hand bat        Legbreak googly      Australia    Player   

   ...  City_Name Country_Name_y                  Toss_Winner  \
0  ...  Bangalore          India  Royal Challengers Bangalore   
1  ...  Bangalore          India  Royal Challengers Bangalore   
2  ...  Bangalore          India  Royal Challengers Bangalore   
3  ...  Bangalore          India  Royal Challengers Bangalore   
4  ...  Bangalore          India  Royal Challengers Bangalore   

            match_winner  Toss_Name  Win_Type Outcome_Type    ManOfMach  \
0  Kolkata Knight Riders      field      runs       Result  BB McCullum   
1  Kolkata Knight Riders      field      runs       Result  BB McCullum   
2  Kolkata Knight Riders      field      runs       Result  BB McCullum   
3  Kolkata Knight Riders      field      runs       Result  BB McCullum   
4  Kolkata Knight Riders      field      runs       Result  BB McCullum   

  Win_Margin Country_id  
0      140.0          1  
1      140.0          1  
2      140.0          1  
3      140.0          1  
4      140.0          1  

[5 rows x 36 columns]

In [7]:
clean_df['Bowling_skill'].fillna('not a baller', inplace=True)
null_bowling_df = clean_df[clean_df['Bowling_skill'].isnull()]
null_bowling_df
#the data points where bowling skill is null, that player is not a baller, 
#and so we fill the data points with 'not a baller'

Empty DataFrame
Columns: [Player_match_SK, PlayerMatch_key, match_id, Player_Id, Player_Name, DOB, Batting_hand, Bowling_skill, Country_Name_x, Role_Desc, Player_team, Opposit_Team, Season_year, is_manofThematch, Age_As_on_match, IsPlayers_Team_won, Player_Captain, Opposit_captain, Player_keeper, Opposit_keeper, Match_SK, Team1, Team2, match_date, Season_Year, Venue_Name, City_Name, Country_Name_y, Toss_Winner, match_winner, Toss_Name, Win_Type, Outcome_Type, ManOfMach, Win_Margin, Country_id]
Index: []

[0 rows x 36 columns]

In [8]:
clean_df['match_winner'].fillna('Match Abandoned', inplace=True)
clean_df['Win_Type'].fillna('Match Tied', inplace=True)
clean_df['ManOfMach'].fillna('No Man of the match', inplace=True)
clean_df['Win_Margin'].fillna('Match abandoned or Tied', inplace=True)
clean_df.head()

Player_match_SK  PlayerMatch_key  match_id  Player_Id Player_Name  \
0            12694      33598700006    335987          6    R Dravid   
1            12695      33598700007    335987          7    W Jaffer   
2            12696      33598700008    335987          8     V Kohli   
3            12697      33598700009    335987          9   JH Kallis   
4            12698      33598700010    335987         10    CL White   

          DOB    Batting_hand          Bowling_skill Country_Name_x Role_Desc  \
0   1/11/1973  Right-hand bat     Right-arm offbreak          India   Captain   
1   2/16/1978  Right-hand bat     Right-arm offbreak          India    Player   
2   11/5/1988  Right-hand bat       Right-arm medium          India    Player   
3  10/16/1975  Right-hand bat  Right-arm fast-medium   South Africa    Player   
4   8/18/1983  Right-hand bat        Legbreak googly      Australia    Player   

   ...  City_Name Country_Name_y                  Toss_Winner  \
0  ...  Bangalore          India  Royal Challengers Bangalore   
1  ...  Bangalore          India  Royal Challengers Bangalore   
2  ...  Bangalore          India  Royal Challengers Bangalore   
3  ...  Bangalore          India  Royal Challengers Bangalore   
4  ...  Bangalore          India  Royal Challengers Bangalore   

            match_winner  Toss_Name  Win_Type Outcome_Type    ManOfMach  \
0  Kolkata Knight Riders      field      runs       Result  BB McCullum   
1  Kolkata Knight Riders      field      runs       Result  BB McCullum   
2  Kolkata Knight Riders      field      runs       Result  BB McCullum   
3  Kolkata Knight Riders      field      runs       Result  BB McCullum   
4  Kolkata Knight Riders      field      runs       Result  BB McCullum   

  Win_Margin Country_id  
0      140.0          1  
1      140.0          1  
2      140.0          1  
3      140.0          1  
4      140.0          1  

[5 rows x 36 columns]

Data by this point is completely clean. 
There were 4 columns that would've been major factors in analyzing the data, and they had null values.
1. Bowling skill : Since every data point is essential in our dataset, this column had 1130 values that were null. We researched on google diving into the details of the actual matches and figured out that these players who did not have a bowling skill were non-ballers. This means, they were either batters or fielders or wicket-keepers. To overcome this, we changed the null values to 'not a baller'.
2. Match winner : Since every match is important for our sport analysis, this column had 66 null values. Upon researching on google, we figured out that these matches did not have a winner due to match getting abandoned for reasons concerned with weather conditions. To fill in these gaps, we changed the null values to 'match abandoned'.
   Reference link for the reserach : https://www.espncricinfo.com/series/indian-premier-league-2011-466304/delhi-daredevils-vs-pune-warriors-68th-match-501265/match-report
3. Win type : The null values in these columns were for matches that were neither won nor lost. These matches were tied. So, we filled in the null values with 'match tied'.
   Reference link for the research : https://www.cricketwa.com/article/172/super-over-in-ipl.aspx
4. Win margin : Win margins are based on whether the winning team won the match by extra runs, or extra wickets. The null values in this column substituted to the matches that were either tied or abandoned. So to fill in the gaps, we changed the null values to 'match abandoned or tied'.
   Reference link for research : https://www.cricketwa.com/article/172/super-over-in-ipl.aspx (tied matches)
          https://www.espncricinfo.com/series/indian-premier-league-2011-466304/delhi-daredevils-vs-pune-warriors-68th-match-501265/match-report (match abandoned)

Our data is entirely clean now for further analysis and we are ready to answer the questions that we initially set up for the analysis.

In [9]:
# Count the number of null values in each column
null_count1 = clean_df.isnull().sum()

# Print the results
print(null_count1)

Player_match_SK       0
PlayerMatch_key       0
match_id              0
Player_Id             0
Player_Name           0
DOB                   0
Batting_hand          0
Bowling_skill         0
Country_Name_x        0
Role_Desc             0
Player_team           0
Opposit_Team          0
Season_year           0
is_manofThematch      0
Age_As_on_match       0
IsPlayers_Team_won    0
Player_Captain        0
Opposit_captain       0
Player_keeper         0
Opposit_keeper        0
Match_SK              0
Team1                 0
Team2                 0
match_date            0
Season_Year           0
Venue_Name            0
City_Name             0
Country_Name_y        0
Toss_Winner           0
match_winner          0
Toss_Name             0
Win_Type              0
Outcome_Type          0
ManOfMach             0
Win_Margin            0
Country_id            0
dtype: int64


In [10]:
clean_df.columns

Index(['Player_match_SK', 'PlayerMatch_key', 'match_id', 'Player_Id',
       'Player_Name', 'DOB', 'Batting_hand', 'Bowling_skill', 'Country_Name_x',
       'Role_Desc', 'Player_team', 'Opposit_Team', 'Season_year',
       'is_manofThematch', 'Age_As_on_match', 'IsPlayers_Team_won',
       'Player_Captain', 'Opposit_captain', 'Player_keeper', 'Opposit_keeper',
       'Match_SK', 'Team1', 'Team2', 'match_date', 'Season_Year', 'Venue_Name',
       'City_Name', 'Country_Name_y', 'Toss_Winner', 'match_winner',
       'Toss_Name', 'Win_Type', 'Outcome_Type', 'ManOfMach', 'Win_Margin',
       'Country_id'],
      dtype='object')

# Team Performance Analysis

In [11]:
TeamP_df = clean_df[['match_id', 'Team1', 'Team2', 'Win_Type', 'match_winner', 'Win_Margin', 'Season_Year']]
TeamP_df.set_index('match_id', inplace=True)
TeamP_df

Team1                    Team2 Win_Type  \
match_id                                                                  
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
...                               ...                      ...      ...   
1082650                Mumbai Indians  Rising Pune Supergiants      run   
1082650                Mumbai Indians  Rising Pune Supergiants      run   
1082650                Mumbai Indians  Rising Pune Supergiants      run   
1082650                Mumbai Indians  Rising Pune Supergiants      run   
1082650                Mumbai Indians  Rising Pune Supergiants      run   

                   match_winner Win_Margin  Season_Year  
match_id                                                 
335987    Kolkata Knight Riders      140.0         2008  
335987    Kolkata Knight Riders      140.0         2008  
335987    Kolkata Knight Riders      140.0         2008  
335987    Kolkata Knight Riders      140.0         2008  
335987    Kolkata Knight Riders      140.0         2008  
...                         ...        ...          ...  
1082650          Mumbai Indians        1.0         2017  
1082650          Mumbai Indians        1.0         2017  
1082650          Mumbai Indians        1.0         2017  
1082650          Mumbai Indians        1.0         2017  
1082650          Mumbai Indians        1.0         2017  

[13992 rows x 6 columns]

In [12]:
# Count the number of null values in each column
null_count1 = TeamP_df.isnull().sum()

# Print the results
print(null_count1)

Team1           0
Team2           0
Win_Type        0
match_winner    0
Win_Margin      0
Season_Year     0
dtype: int64


In [13]:
TeamP_df['Win_Type'] = TeamP_df['Win_Type'].replace(0, "tie or abandoned")
TeamP_df['Win_Type'] = TeamP_df['Win_Type'].replace("run", "runs")
TeamP_df['Win_Margin'] = TeamP_df['Win_Margin'].replace("Match abandoned or Tied", 0)
TeamP_df

/tmp/ipykernel_55/1936598604.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  TeamP_df['Win_Type'] = TeamP_df['Win_Type'].replace(0, "tie or abandoned")
/tmp/ipykernel_55/1936598604.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  TeamP_df['Win_Type'] = TeamP_df['Win_Type'].replace("run", "runs")
/tmp/ipykernel_55/1936598604.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the d

Team1                    Team2 Win_Type  \
match_id                                                                  
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
335987    Royal Challengers Bangalore    Kolkata Knight Riders     runs   
...                               ...                      ...      ...   
1082650                Mumbai Indians  Rising Pune Supergiants     runs   
1082650                Mumbai Indians  Rising Pune Supergiants     runs   
1082650                Mumbai Indians  Rising Pune Supergiants     runs   
1082650                Mumbai Indians  Rising Pune Supergiants     runs   
1082650                Mumbai Indians  Rising Pune Supergiants     runs   

                   match_winner  Win_Margin  Season_Year  
match_id                                                  
335987    Kolkata Knight Riders       140.0         2008  
335987    Kolkata Knight Riders       140.0         2008  
335987    Kolkata Knight Riders       140.0         2008  
335987    Kolkata Knight Riders       140.0         2008  
335987    Kolkata Knight Riders       140.0         2008  
...                         ...         ...          ...  
1082650          Mumbai Indians         1.0         2017  
1082650          Mumbai Indians         1.0         2017  
1082650          Mumbai Indians         1.0         2017  
1082650          Mumbai Indians         1.0         2017  
1082650          Mumbai Indians         1.0         2017  

[13992 rows x 6 columns]

In [16]:
grouped_df = TeamP_df.groupby(['match_winner', 'Season_Year'])['Win_Margin'].count().div(22)
grouped_df.head(25)

match_winner         Season_Year
Chennai Super Kings  2008            9.0
                     2009            8.0
                     2010            9.0
                     2011           11.0
                     2012           10.0
                     2013           12.0
                     2014           10.0
                     2015           10.0
Deccan Chargers      2008            2.0
                     2009            9.0
                     2010            8.0
                     2011            6.0
                     2012            4.0
Delhi Daredevils     2008            7.0
                     2009           10.0
                     2010            7.0
                     2011            4.0
                     2012           11.0
                     2013            3.0
                     2014            2.0
                     2015            5.0
                     2016            7.0
                     2017            6.0
Gujarat Lions        201

# Venue Analysis

In [ ]:
# Group the dataset by venue and season and calculate the number of matches played, wins and win percentage
#venue_performance = clean_df.groupby(['Venue_Name', 'Season_Year']).agg({'match_id': 'nunique',
#                                                                         'match_winner': lambda x: (x == clean_df['Venue_Name']).sum(),
#                                                                         'Win_Type': lambda x: (x == 'by runs').sum() + (x == 'by wickets').sum(),
#                                                                         'Win_Type': 'count'})


In [ ]:
# Rename columns
#venue_performance = venue_performance.rename(columns={'match_id': 'Matches_Played', 
#                                                      '<lambda_0>': 'Matches_Won', 
#                                                      'Win_Type': 'Matches_Win_Type',
#                                                      '<lambda_1>': 'Matches_Total'})

# Calculate win percentage
#venue_performance['Win_Percentage'] = (venue_performance['Matches_Won'] / venue_performance['Matches_Played']) * 100

# Sort the data by win percentage
#venue_performance = venue_performance.sort_values(by='Win_Percentage', ascending=False)

# Print the first 10 rows of the dataframe
#print(venue_performance.head(10))


In [ ]:
#venue_performance = clean_df.groupby(['Venue_Name', 'Season_Year']).agg({
#    'match_id': 'count',
#    'Win_Type': 'count',
#    'match_winner': 'count',
#    'Win_Margin': 'sum'
#})
